tests/controller/insertcontent.js

```diff
@@ -62,6 +62,57 @@ describe( 'DataController', () => {
 			expect( getData( doc ) ).to.equal( 'xa[]x' );
 		} );
 
+		describe( 'saves the reference to the original object', () => {
+			it( 'inline-widget', () => {
+				doc = new Document();
+				doc.createRoot();
+
+				doc.schema.registerItem( 'paragraph', '$block' );
+				doc.schema.registerItem( 'inlineWidget' );
+				doc.schema.allow( { name: 'inlineWidget', inside: '$block' } );
+				doc.schema.allow( { name: 'inlineWidget', inside: '$clipboardHolder' } );
+				doc.schema.objects.add( 'inlineWidget' );
+
+				dataController = new DataController( doc );
+
+				const batch = doc.batch();
+
+				setData( doc, '<paragraph>f[]oo</paragraph>' );
+
+				const content = parse( '<inlineWidget></inlineWidget>', doc.schema, {
+					context: [ '$clipboardHolder' ]
+				} );
+
+				insertContent( dataController, content, doc.selection, batch );
+
+				expect( doc.getRoot().getChild( 0 ).getChild( 1 ) ).to.equal( content );
+			} );
+
+			it( 'image', () => {
```

Comment:
No, in the first one I'm inserting inlineWidget and in second image, but I can remove the first one of course if only the second one is necessary.